### PR TITLE
[Silabs] - Updating the wifi-sdk to v3.1.1 with required changes

### PR DESCRIPTION
**Pull request by silabs-srishylam**
* Updating the wifi_sdk to v3.1.1
* Replace the macros which got changed from wifi_sdk
* Updated the paths in GNI with new wifi-sdk
* Aligned the linker file with the wifi_sdk platform
* Updated the API's which got modified in new wifi_sdk

### CODE REVIEW

File: integrations/docker/images/stage-2/chip-build-efr32/Dockerfile [hunk 1]
```diff
@@ -30,8 +30,8 @@ RUN git clone --depth=1 --single-branch --branch=2.8.2 https://github.com/Silico
     rm -rf .git \
     && : # last line
 
-# Clone WiSeConnect SDK 3.1.0 (58a2404)
-RUN git clone --depth=1 --single-branch --branch=v3.1.0 https://github.com/SiliconLabs/wiseconnect.git /tmp/wifi_sdk && \
+# Clone WiSeConnect SDK 3.1.1 (b2c1cd0)
```
Comment (jmartinez-silabs) [inline integrations/docker/images/stage-2/chip-build-efr32/Dockerfile:33]: If you are updating the docker file, it is required you update the docker image version. 

That also confuse me that the CI is passing. As this new docker image doesn't exist yet for the CI. Are you not using wise connect for docker for the CI? If so then why add it to the docker file?


Usually the docker update is done in a separate PR that needs to be merged before the one that updates the code. Else the docker image isn't yet available for the CI

File: src/platform/silabs/BLEManagerImpl.h [hunk 2]
```diff
@@ -25,15 +25,15 @@
 #pragma once
 #if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
 
-#if (BLE_ENABLE || RSI_BLE_ENABLE)
+#if (SLI_SI91X_ENABLE_BLE || RSI_BLE_ENABLE)
 #define BLE_MIN_CONNECTION_INTERVAL_MS 45 // 45 msec
 #define BLE_MAX_CONNECTION_INTERVAL_MS 45 // 45 msec
 #define BLE_SLAVE_LATENCY_MS 0
 #define BLE_TIMEOUT_MS 400
 #endif // RSI_BLE_ENABLE
```
Comment (mkardous-silabs) [inline src/platform/silabs/BLEManagerImpl.h:33]: comment needs to be update here and for the #endif associated with the #if of line 36

File: src/platform/silabs/rs911x/rsi_ble_config.h [hunk 3]
```diff
@@ -89,7 +89,7 @@
 #define RSI_BLE_CUSTOM_SERVICE_UUID (0xFFF6)
 #define RSI_BLE_CUSTOM_LEVEL_UUID (0x1FF1)
 
-#ifdef RSI_M4_INTERFACE
+#if (defined(RSI_M4_INTERFACE) || defined(SLI_SI91X_MCU_INTERFACE))
```
Comment (mkardous-silabs) [inline src/platform/silabs/rs911x/rsi_ble_config.h:92]: #else or #endif needs to be update here as well.

File: src/platform/silabs/BLEManagerImpl.h (hunk 2 above)
Comment (arun-silabs) [reply src/platform/silabs/BLEManagerImpl.h]: updated comments.

File: src/platform/silabs/rs911x/rsi_ble_config.h (hunk 3 above)
Comment (arun-silabs) [reply src/platform/silabs/rs911x/rsi_ble_config.h:92]: updated comments in else and endif statements

File: integrations/docker/images/stage-2/chip-build-efr32/Dockerfile [hunk 4]
```diff
@@ -30,8 +30,8 @@ RUN git clone --depth=1 --single-branch --branch=2.8.2 https://github.com/Silico
     rm -rf .git \
     && : # last line
 
-# Clone WiSeConnect SDK 3.1.1 (b2c1cd0)
```
Comment (chirag-silabs) [inline integrations/docker/images/stage-2/chip-build-efr32/Dockerfile:33]: This file shouldn't be changed.

Comment (srningap) [reply integrations/docker/images/stage-2/chip-build-efr32/Dockerfile:33]: ohhh! This got changed with rebase, I will revert it.

File: third_party/silabs/SiWx917_sdk.gni [hunk 5]
```diff
@@ -54,45 +47,43 @@ template("siwx917_sdk") {
       "${efr32_sdk_root}/platform/emdrv/common/inc",
       "${efr32_sdk_root}/platform/service/device_init/inc",
       "${sdk_support_root}/matter/mbedtls/tinycrypt/inc",
-      "${sdk_support_root}/matter/si91x/siwx917/${sdk_support_board}/autogen",
-      "${sdk_support_root}/matter/si91x/siwx917/${sdk_support_board}/config",
+      "${sdk_support_root}/matter/si91x/siwx917/BRD4338A/autogen",
```
Comment (jmartinez-silabs) [inline third_party/silabs/SiWx917_sdk.gni:50]: ok for now but it could be more future-proof if we use ${silabs_board}

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by jmartinez-silabs** [comment]
For the buid error with efr32, you will need #31264

**Review by saurabhst — APPROVED** [review]

**Pull request merged**